### PR TITLE
Make HTTP connection TTL configurable

### PR DESCRIPTION
**Pull request by kerkhofsd**

### CODE REVIEW

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java [hunk 1]
```diff
@@ -11,7 +12,9 @@
 public class InsecureSslHttpComponentsClientHttpRequestFactory extends HttpComponentsClientHttpRequestFactory {
 
     @SneakyThrows
-    public InsecureSslHttpComponentsClientHttpRequestFactory() {
+    public InsecureSslHttpComponentsClientHttpRequestFactory(HttpClient httpClient) {
+        super(httpClient);
```
Comment (tgeens) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java]: Wait, you are setting/passing an `HttpClient` here, but a few lines down, we create and set a new one with `this.setHttpClient(...)` ?

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java [hunk 2]
```diff
@@ -22,4 +25,8 @@ public InsecureSslHttpComponentsClientHttpRequestFactory() {
                 .setSSLHostnameVerifier(new NoopHostnameVerifier())
```
Comment (tgeens) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java]: Can't the TTL be configured here ?
`.setConnectionTimeToLive(httpProperties.getConnectionTimeToLive(), TimeUnit.MILLISECONDS)`

**Review by tgeens — CHANGES_REQUESTED** [review]

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java (hunk 1 above)
Comment (kerkhofsd) [reply alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java]: Good point, I overlooked that.

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java [hunk 3]
```diff
@@ -12,9 +12,7 @@
 public class InsecureSslHttpComponentsClientHttpRequestFactory extends HttpComponentsClientHttpRequestFactory {
 
     @SneakyThrows
-    public InsecureSslHttpComponentsClientHttpRequestFactory(HttpClient httpClient) {
-        super(httpClient);
-
+    public InsecureSslHttpComponentsClientHttpRequestFactory(long connectionTimeToLive) {
```
Comment (tgeens) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/http/InsecureSslHttpComponentsClientHttpRequestFactory.java]: it's a bit silly adding a single configuration-parameter here, isn't it ?

Alternatives:
1. pass `HttpProperties` along ?
2. what about allowing a callback with `Consumer<HttpClientBuilder>`, so the caller can configure the http client at will ?

I have a personal preference for (2)

**Review by tgeens — APPROVED** [review]

**Pull request merged**